File: C_Programming/Competitive_Programming/CP_020.c
//////////////////////////////////////////////////////////////////
//
//  File Name    : Difference_Between_Factors_And_NonFactors
//
//  Description  : This program accepts a number from the user
//                 and returns the difference between the
//                 summation of its factors and the summation
//                 of its non-factors.
//
//  Example:
//      Input  : 12
//      Output : -34
//
//      Input  : 10
//      Output : -29
//
//////////////////////////////////////////////////////////////////

#include<stdio.h>

int FactDiff(int iNo)
{
    int iCnt = 0;
    int iAddFact = 0;
    int iAddNonFact = 0;

    for(iCnt = 1; iCnt < iNo; iCnt++)
    {
        if(iNo % iCnt == 0)
        {
            iAddFact = iAddFact + iCnt;
        }
        else
        {
            iAddNonFact = iAddNonFact + iCnt;
        }
    }
    
    return (iAddFact - iAddNonFact);
}

int main()
{
    int iValue = 0;
    int iRet = 0;

    printf("Enter Number : \n");
    scanf("%d", &iValue);

    iRet = FactDiff(iValue);

    printf("%d", iRet);

    return 0;
}

File: Java_Programming/Competitive_Programming/CP_020.java
//////////////////////////////////////////////////////////////////
//
//  File Name    : Find_Smallest_Digit
//
//  Description  : This program accepts a number and
//                 finds the smallest digit present
//                 in the given number.
//
//  Example:
//      Input  : 45872
//      Output : Smallest Digit is : 2
//
//////////////////////////////////////////////////////////////////

class Logic
{
    void findSmallestDigit(int iNum)
    {
        int iDigit = 0, iComp = 0;
        int iNo = iNum;

        while(iNo > 0)  
        {
            iDigit = iNo % 10; 

            if(iComp == 0)
            {
                iComp = iDigit;
            }

            if(iComp > iDigit)  
            {
                iComp = iDigit;  
            }
            iNo = iNo / 10;  
        }
        System.out.println("Largest Digit is : " + iComp);
    }
}

class CP_020
{
    public static void main(String A[])
    {
        Logic lobj = new Logic();

        lobj.findSmallestDigit(45872);
    }
}